src/main/java/CLI/Main.java

```diff
@@ -375,4 +387,91 @@ private static void checkHistoryAddJobsAndDownload(Job job, boolean removeInputB
             downloader.run();
         }
     }
-}
+
+    public static boolean checkUpdate() throws URISyntaxException {
+        String latestVersion = getLatestVersion();
+        if (isNewerVersion(latestVersion , VERSION_NUMBER)) {
+            String Link;
+            String oldFilePath = String.valueOf(Main.class.getProtectionDomain().getCodeSource().getLocation().toURI());
+            String newFilePath = "";
+            String OS_NAME = OS.getOSName();
+            if (OS_NAME.contains("win")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI.exe";
+                String fileName = "Drifty_CLI.exe";
+                String dirPath = String.valueOf(Paths.get(System.getenv("LOCALAPPDATA"), "Drifty", "updates"));
+                FileDownloader downloader =  new FileDownloader(Link , fileName , dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            } else if (OS_NAME.contains("mac")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI_macos";
+                String fileName = "Drifty-CLI_macos";
+                String dirPath = ".drifty/updates";
+                FileDownloader downloader =  new FileDownloader(Link , fileName , dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            } else if (OS_NAME.contains("linux")) {
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI_linux";
+                String fileName = "Drifty-CLI_linux";
+                String dirPath = ".drifty/updates";
+                FileDownloader downloader = new FileDownloader(Link, fileName, dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            }
+            Updater.replaceUpdate(oldFilePath , newFilePath);
```

Comment:
@kmr-rohit The updater will be a completely separate executable and it will be downloaded once the updated Drifty executable has been downloaded.
The old Path and the path of the new executable will be passed as Command-line argument to the updater which will just replace and re-launch the updated executable.

Reply:
Implemented separate Package, where old Path and New Path will be passed as Command-line Arguments.

Reply:
Okay :+1: